pkg/logger/logger.go

```diff
@@ -0,0 +1,75 @@
+// Copyright © 2019 The Homeport Team
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package logger
+
+//Logger defines an observable logger
+type Logger interface {
+	//GetChannelProvider returns the channel provider this Cluster uses
```

Comment:
In Godoc, the comments above an interface function (sometimes called method) are not evaluated as a special thing in the generated documentation. Even though you find well known package that do this kind of commenting, I would suggest to actually put the comments into the Logger interface comment body with multiple lines and empty lines for readability.

Reply:
As I followed https://blog.golang.org/godoc-documenting-go-code as a guide on how to comment on Golang, it states that documentation convention "is to document a type, variable, constant, function, or even a package, write a regular comment directly preceding its declaration". If this is not a ground breaking issue, I would like to keep the methods themselves documented instead of the actual interface containing the docs of all the defined functions.